File: codegen/results/v002/m3-max-48gb/lmstudio/qwen_qwen3.8-27b-q4_k_m_nothink/20260829_173518/python/007_beatles_interview/solution.py
import csv
import json
from datetime import date

def calculate_age(birthday, as_of):
    """
    Calculate age as of a given date.
    If the person has died before the as_of date, calculate age at death.
    Otherwise, calculate age as of the as_of date.
    """
    birthday_date = datetime.datetime.strptime(birthday, "%m/%d/%Y").date()
    if died:
        death_date = datetime.datetime.strptime(died, "%m/%d/%Y").date()
        if death_date < as_of:
            return (death_date.year - birthday_date.year) - ((death_date.month, death_date.day) < (birthday_date.month, birthday_date.day))
    # Calculate age as of as_of date
    if (as_of.month, as_of.day) < (birthday_date.month, birthday_date.day):
        return as_of.year - birthday_date.year - 1
    else:
        return as_of.year - birthday_date.year

import datetime

as_of = date(2025, 7, 1)
results = []

with open('input/input.csv', 'r') as f:
    reader = csv.DictReader(f)
    for row in reader:
        # Parse name
        parts = row['Name'].split()
        first_name = parts[0]
        last_name = parts[-1]
        
        birthday = row['Birthday']
        died = row['Died']
        
        # Calculate age
        birthday_date = datetime.datetime.strptime(birthday, "%m/%d/%Y").date()
        if died and died != 'null':
            death_date = datetime.datetime.strptime(died, "%m/%d/%Y").date()
            if death_date < as_of:
                # Age at death
                age = (death_date.year - birthday_date.year) - ((death_date.month, death_date.day) < (birthday_date.month, birthday_date.day))
            else:
                # Age as of as_of date (if died after as_of or on as_of, treat as alive on as_of)
                age = (as_of.year - birthday_date.year) - ((as_of.month, as_of.day) < (birthday_date.month, birthday_date.day))
        else:
            # Alive on as_of date
            age = (as_of.year - birthday_date.year) - ((as_of.month, as_of.day) < (birthday_date.month, birthday_date.day))
        
        # Build relatives list
        relatives = []
        
        father = row['Father']
        if father and father != 'null':
            father_parts = father.split()
            relatives.append({
                "FirstName": father_parts[0],
                "LastName": father_parts[-1],
                "Relationship": "Father"
            })
        
        mother = row['Mother']
        if mother and mother != 'null':
            mother_parts = mother.split()
            relatives.append({
                "FirstName": mother_parts[0],
                "LastName": mother_parts[-1],
                "Relationship": "Mother"
            })
        
        brother = row['Brother']
        if brother and brother != 'null':
            brother_parts = brother.split()
            relatives.append({
                "FirstName": brother_parts[0],
                "LastName": brother_parts[-1],
                "Relationship": "Brother"
            })
        
        sister = row['Sister']
        if sister and sister != 'null':
            sister_parts = sister.split()
            relatives.append({
                "FirstName": sister_parts[0],
                "LastName": sister_parts[-1],
                "Relationship": "Sister"
            })
        
        # Format birthday as YYYY-MM-DD
        birthday_formatted = datetime.datetime.strptime(birthday, "%m/%d/%Y").strftime("%Y-%m-%d")
        
        results.append({
            "FirstName": first_name,
            "LastName": last_name,
            "Birthday": birthday_formatted,
            "Age": age,
            "Relatives": relatives
        })

print(json.dumps(results, indent=2))

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen_qwen3.8-27b-q4_k_m_nothink/20260829_173518/go/007_beatles_interview/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strings"
	"time"
)

type Relative struct {
	FirstName    string `json:"FirstName"`
	LastName     string `json:"LastName"`
	Relationship string `json:"Relationship"`
}

type Person struct {
	FirstName string     `json:"FirstName"`
	LastName  string     `json:"LastName"`
	Birthday  string     `json:"Birthday"`
	Age       int        `json:"Age"`
	Relatives []Relative `json:"Relatives"`
}

func main() {
	file, err := os.Open("input/input.csv")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	reader := csv.NewReader(file)
	records, err := reader.ReadAll()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading CSV: %v\n", err)
		os.Exit(1)
	}

	// Skip header
	if len(records) < 2 {
		fmt.Println("[]")
		return
	}

	var people []Person
	refDate := time.Date(2025, 7, 1, 0, 0, 0, 0, time.UTC)

	for _, record := range records[1:] {
		if len(record) < 7 {
			continue
		}
		
		name := strings.TrimSpace(record[0])
		parts := strings.Fields(name)
		var firstName, lastName string
		if len(parts) >= 2 {
			firstName = parts[0]
			lastName = parts[len(parts)-1]
		} else if len(parts) == 1 {
			firstName = parts[0]
			lastName = ""
		}
		
		birthdayStr := strings.TrimSpace(record[1])
		birthday, err := time.Parse("1/2/2006", birthdayStr)
		if err != nil {
			continue
		}
		
		age := refDate.Year() - birthday.Year()
		if refDate.YearDay() < birthday.YearDay() {
			age--
		}
		
		var relatives []Relative
		
		father := strings.TrimSpace(record[3])
		if father != "null" && father != "" {
			fParts := strings.Fields(father)
			if len(fParts) >= 2 {
				relatives = append(relatives, Relative{
					FirstName:    fParts[0],
					LastName:     fParts[len(fParts)-1],
					Relationship: "Father",
				})
			} else if len(fParts) == 1 {
				relatives = append(relatives, Relative{
					FirstName:    fParts[0],
					LastName:     "",
					Relationship: "Father",
				})
			}
		}
		
		mother := strings.TrimSpace(record[4])
		if mother != "null" && mother != "" {
			mParts := strings.Fields(mother)
			if len(mParts) >= 2 {
				relatives = append(relatives, Relative{
					FirstName:    mParts[0],
					LastName:     mParts[len(mParts)-1],
					Relationship: "Mother",
				})
			} else if len(mParts) == 1 {
				relatives = append(relatives, Relative{
					FirstName:    mParts[0],
					LastName:     "",
					Relationship: "Mother",
				})
			}
		}
		
		brother := strings.TrimSpace(record[5])
		if brother != "null" && brother != "" {
			bParts := strings.Fields(brother)
			if len(bParts) >= 2 {
				relatives = append(relatives, Relative{
					FirstName:    bParts[0],
					LastName:     bParts[len(bParts)-1],
					Relationship: "Brother",
				})
			} else if len(bParts) == 1 {
				relatives = append(relatives, Relative{
					FirstName:    bParts[0],
					LastName:     "",
					Relationship: "Brother",
				})
			}
		}
		
		sister := strings.TrimSpace(record[6])
		if sister != "null" && sister != "" {
			sParts := strings.Fields(sister)
			if len(sParts) >= 2 {
				relatives = append(relatives, Relative{
					FirstName:    sParts[0],
					LastName:     sParts[len(sParts)-1],
					Relationship: "Sister",
				})
			} else if len(sParts) == 1 {
				relatives = append(relatives, Relative{
					FirstName:    sParts[0],
					LastName:     "",
					Relationship: "Sister",
				})
			}
		}
		
		if relatives == nil {
			relatives = []Relative{}
		}
		
		people = append(people, Person{
			FirstName: firstName,
			LastName:  lastName,
			Birthday:  birthday.Format("2006-01-02"),
			Age:       age,
			Relatives: relatives,
		})
	}
	
	if people == nil {
		people = []Person{}
	}
	
	output, err := json.MarshalIndent(people, "", "  ")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error marshaling JSON: %v\n", err)
		os.Exit(1)
	}
	
	fmt.Println(string(output))
}